src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataReaderExtensions.cs

```diff
@@ -158,12 +158,26 @@ private static void ReadCollectionItem(ODataReader reader, Stack<ODataItemBase>
                 case ODataReaderState.EntityReferenceLink:
                     ODataEntityReferenceLink entityReferenceLink = (ODataEntityReferenceLink)reader.Item;
                     Contract.Assert(entityReferenceLink != null, "Entity reference link should never be null.");
-                    ODataEntityReferenceLinkBase entityReferenceLinkWrapper = new ODataEntityReferenceLinkBase(entityReferenceLink);
+                    ODataEntityReferenceLinkWrapper entityReferenceLinkWrapper = new ODataEntityReferenceLinkWrapper(entityReferenceLink);
 
                     Contract.Assert(itemsStack.Count > 0, "Entity reference link should never be reported as top-level item.");
                     {
                         ODataNestedResourceInfoWrapper parentNavigationLink = (ODataNestedResourceInfoWrapper)itemsStack.Peek();
                         parentNavigationLink.NestedItems.Add(entityReferenceLinkWrapper);
+                        
+                        parentNavigationLink.AppendReferenceLink(entityReferenceLinkWrapper);
+
+                        if (parentNavigationLink.NestedLinks.Count > 1)
+                        {
+                            ODataEntityReferenceLinksWrapper entityReferenceLinksWrapper = new ODataEntityReferenceLinksWrapper(null);
+                            foreach (ODataEntityReferenceLinkWrapper dataEntityReferenceLinkWrapper in parentNavigationLink.NestedLinks)
+                            {
+                                entityReferenceLinksWrapper.EntityReferenceLinks.Add(dataEntityReferenceLinkWrapper);
+                                parentNavigationLink.NestedItems.Remove(dataEntityReferenceLinkWrapper);
+                            }
+
+                            parentNavigationLink.NestedItems.Add(entityReferenceLinksWrapper);
+                        }
```

Comment:
append to NesteItems

Reply:
@xuzhg you mean  parentNavigationLink.NestedItems.Append(entityReferenceLinksWrapper);